### PR TITLE
Update the concepts page on FittingMinimizers

### LINKED ISSUE #17594: Update the concepts page on FittingMinimizers with more minimizers and datasets
This concerns several additions to the [minimizers page](http://docs.mantidproject.org/nightly/concepts/FittingMinimizers.html
) that should be included in the upcoming release 3.8:
- Include the "Trust region" minimizer (the one currently called DTRS) as a new alternative (one new column in the tables).
- In addition to the traditional NIST benchmarks/tables included in release 3.7, include comparison tables for neutron datasets (and less important, some CUTEst problems).

The tables are generated with the scripts being added in #15952.

### PR DESCRIPTION
**Pull request by FedeMPouzols**
Adds new comparisons: 
- The "Trust region" minimizer
- Tables with results for neutron data and "CUTEst" problems.

**To test:**
- Build the documentation (Qt help or HTML) and check the concep page `FittingMinimizers`
- Check that the new tables make sense. Feedback needed!
- Make sure that there are no details missing.

(Find the problem/data files zip that is used by the scripts here (at ISIS): \\olympic\Babylon5\Public\fedemp\fitting\FittingTestData.zip)

Fixes #17594.

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [x] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [x] Has the relevant documentation been added/updated?
- [x] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by FedeMPouzols** [comment]
@Anders-Markvardsen, @tom-perkins: this is not ready for testing yet (I have to fix a few links between the tables). But you can already see some results for the trust region/DTRS minimizer and some neutron data problems. I expect to have this ready tomorrow morning.

**Comment by FedeMPouzols** [comment]
@Anders-Markvardsen, @tom-perkins: this is now ready for a check. There is still one issue with the links from table cells -> detailed results which are broken at the moment, so you may need to scroll up/down the detailed tables to find your specific results. But other than that I think that all the relevant information is already there. I'll try to fix the scripts to generate the links properly today in between meetings.

File: docs/source/concepts/FittingMinimizersComparisonDetailed.rst [hunk 1]
```diff
@@ -80,14 +88,24 @@ weighted least squares as cost function
...
 Run time for individual NIST problems, "higher" difficulty
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-.. include:: minimizers_comparison/v3.7.0/comparison_unweighted_v3.7_runtime_nist_higher.txt
+.. include:: minimizers_comparison/v3.8.0/comparison_unweighted_v3.8_runtime_nist_higher.txt
+
+Run time for individual Neutron Data problems
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. include:: minimizers_comparison/v3.8.0/comparison_unweighted_v3.8_runtime_neutron_data.txt
```
Comment (tom-perkins) [inline docs/source/concepts/FittingMinimizersComparisonDetailed.rst]: I see the "Neutron data" header, then the "CUTEst" results table - the table/header in the middle are missing.

File: docs/source/concepts/FittingMinimizersWeightedComparisonDetailed.rst [hunk 2]
```diff
@@ -55,19 +65,19 @@ Comparison in terms of run time
...
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-.. include:: minimizers_comparison/v3.7.0/comparison_weighted_v3.7_runtime_nist_average.txt
+.. include:: minimizers_comparison/v3.8.0/comparison_weighted_v3.8_runtime_nist_average.txt
 
 Run time for individual NIST problems, "higher" difficulty
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-.. include:: minimizers_comparison/v3.7.0/comparison_weighted_v3.7_runtime_nist_higher.txt
+.. include:: minimizers_comparison/v3.8.0/comparison_weighted_v3.8_runtime_nist_higher.txt
```
Comment (tom-perkins) [inline docs/source/concepts/FittingMinimizersWeightedComparisonDetailed.rst]: Are there tables for CUTEst and Neutron Data for this section? (run time comparison, weighted)

**Review by tom-perkins — COMMENTED** [review]
First check - the tables look good and the colours for the results all work well, it just needs the links from summary table cells -> detailed results (as noted above). 

I noticed one table that is present in the code but doesn't appear in the docs when built (`comparison_unweighted_v3.8_runtime_neutron_data`) - not sure why this is!

**Comment by FedeMPouzols** [comment]
Thanks @tom-perkins, yes some tables are missing, and some of them are not correctly parsed by sphinx apparently. I'll see how to fix that.

File: docs/source/concepts/FittingMinimizersWeightedComparisonDetailed.rst (hunk 2 above)
Comment (FedeMPouzols) [inline docs/source/concepts/FittingMinimizersWeightedComparisonDetailed.rst]: Oh yes, thanks for the catch.

**Comment by Anders-Markvardsen** [comment]
My computer just crashed 2 minutes hence it may be specific to my PC but the tables now look like this:

capture

just here checking if this is intentional

**Comment by Anders-Markvardsen** [comment]
The numbers in the summary table for the first set of neutron data are highly interesting, however I currently struggle to understand how they related to the numbers in the detailed table, e.g. the mediam in terms of accuracy is  	1.326e+04 for LM but the (unweighted) detailed table which I have have no number this high..... speak to you guys later...

**Comment by Anders-Markvardsen** [comment]
We should agree to put a few lines on top of the summary table to tell that neutron data may still be classified as work in progress or something equivalent and the same seems to be the case for the new trust region mimimizer

**Comment by FedeMPouzols** [comment]
Is that table screenshot from Internet Explorer? That is a known limitation, and I'm not sure if/when newer versions of IE will work well with this. There is an alternative way of coloring the whole cells that works for IE, but then the tables don't display nicely in the Qt help and other browsers.

**Comment by Anders-Markvardsen** [comment]
I wonder if it something with my development environment then, because I get the same on Chrome, Firefox and Internet explorer, i.e. the table displays as shown above. Also, for these 3 browsers http://docs.mantidproject.org/nightly/concepts/FittingMinimizersComparisonDetailed.html#minimizers-unweighted-comparison-in-terms-of-accuracy as far as I can tell displays correctly - including Internet Explorer 11 - show you tomorrow

**Comment by FedeMPouzols** [comment]
@Anders-Markvardsen, @tom-perkins: the last commits should fix a couple of issues we were just discussiong:
- the issue with the background color of the table cells. 
- The strange median values of the "Neutron data" row" in the summary table. That row is actually CUTEst (the labels were swapped). For CUTEst Damping, Levenberg-Marquardt, Levenberg-MarquardtMD, and Trust Region (mis)behave in a very similar way if you look at the individual results table. That produces the strange pattern by which the L-M and the Trust region columns look the same.  Levenberg-MarquardMD and Damping misbehave similarly but different enough that their medians are different. That's actually interesting as for example for `VESUVIOLS` these two methods have some advantage/good feature that the other generally better methods don't have.

**Comment by tom-perkins** [comment]
The background colours of the table cells now display correctly for me :+1:

**Comment by Anders-Markvardsen** [comment]
All clear on my computer also; in all three browsers and the median numbers

**Comment by Anders-Markvardsen** [comment]
@FedeMPouzols let me know if you would like me to suggest some text, e.g. for the trust-region results with no error

**Comment by FedeMPouzols** [comment]
Great to know! Thanks, I'm adding some text for the few clarifications that we discussed this morning. Feel free to edit them. I'm also fixing some issues with some minimizers and the 'Least squares' cost function which they don't seem to handle well. This was the actual reason behind the fact that some columns were identical in the unweighted tables.

**Comment by FedeMPouzols** [comment]
I think that the tables as modified in the last commit look good now. The problem that we were seeing of poor performance of DTRS with unweighted data was a red herring, a glitch because of a wrong mix of cost function different from 'Least squares` and the `IgnoreInvalidData`. Note how well it performs without exception when not using weights.
The issue with the WISH peaks when using errors came down to the fact that the examples that I got do not calculate any error values and leave them all as 0! We have to see what to do with these tests.

**Comment by Anders-Markvardsen** [comment]
Looks good - this pull-request all clear to me. @tom-perkins what do you think?

One final thing! what about a release note entry something like: "Fit minimizer comparison tables updated to include comparison against neutron data fit examples"

@mantid-roman see the results for trust-region. Let me know if you would me to rename DTRS to TrustRegion? If yes have you created an issue for this already?

**Comment by mantid-roman** [comment]
@Anders-Markvardsen Thanks for reminding me about it. And yes.

**Comment by tom-perkins** [comment]
@Anders-Markvardsen All looks good to me now too.

**Comment by Anders-Markvardsen** [comment]
:shipit: :+1:

**Comment by FedeMPouzols** [comment]
@martyngigg: in case you're going to merge this soon, could you wait 10 mins? I'm going to add better cross-links between tables.

**Comment by martyngigg** [comment]
@FedeMPouzols Sure, no problem.

I'll merge it once you remove the "in progress" label.

**Comment by FedeMPouzols** [comment]
The last commits fix some issues with the links between tables (while going through the sphinx warnings I also realized that one table had gone missing!). The docs build now without any warnings.

**Comment by Anders-Markvardsen** [comment]
Thanks for adding the cross-links from the median table also - I must admit I forgot they where there previously. Apologies for not spotting the missed table also. I now count 10 on the no-error detailed page and 12 on the error-detailed page (which include 2 median tables). Looks even better :shipit:

**Comment by FedeMPouzols** [comment]
Just to keep an eye on potential issues now that the release is so close. There's been a segfault in the doc-tests build: http://builds.mantidproject.org/job/pull_requests-ubuntu/9785/console

**Pull request merged**